cpp/arcticdb/column_store/memory_segment_impl.cpp

```diff
@@ -428,24 +431,42 @@ std::shared_ptr<SegmentInMemoryImpl> SegmentInMemoryImpl::truncate(size_t start_
     auto output = std::make_shared<SegmentInMemoryImpl>();
 
     output->set_row_data(num_values - 1);
-    output->set_string_pool(string_pool_);
     output->set_compacted(compacted_);
+    if (!reconstruct_string_pool) {
+        output->set_string_pool(string_pool_);
+    }
     if (metadata_) {
         google::protobuf::Any metadata;
         metadata.CopyFrom(*metadata_);
         output->set_metadata(std::move(metadata));
     }
 
     for(const auto&& [idx, column] : folly::enumerate(columns_)) {
-        auto truncated_column = Column::truncate(column, start_row, end_row);
-        output->add_column(descriptor_->field(idx), truncated_column);
+        const DataType column_type = column->type().data_type();
+        const Field& field = descriptor_->field(idx);
+        if (is_sequence_type(column_type) && reconstruct_string_pool) {
```

Comment:
I think this logic would be clearer if `Column::truncate` was called before the if statement. Then you can just iterate through `truncated_column` taking a reference to the offset in the column. If it fails the [is_a_string](https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/util/offset_string.hpp#L54) check, leave it as-is. Otherwise, update the value to `output->string_pool().get(string_pool_.get_view(value)).offset()`